python/cmsis_svd/parser.py

```diff
@@ -243,6 +264,54 @@ def duplicate_array_of_registers(input):    #expects a SVDRegister which is an a
             )
         )
     return output
+
+
+
+def propagate_defaults(device):
+    def inherit_register_defaults(source,dest):
+        if source.size is not None:
+            if dest.size is None:
+                dest.size = source.size
+        if source.access is not None:
+            if dest.access is None:
+                dest.access = source.access
+        if source.protection is not None:
+            if dest.protection is None:
+                dest.protection = source.protection
+        if source.reset_value is not None:
+            if dest.reset_value is None:
+                dest.reset_value = source.reset_value
+        if source.reset_mask is not None:
+            if dest.reset_mask is None:
+                dest.reset_mask = source.reset_mask
+
+
+    if device.peripherals:
+        for peripheral in device.peripherals:
+            if peripheral.derived_from:
+                parent = None;
+                for p in device.peripherals:
+                    if p.name == peripheral.derived_from:
+                        parent = p      #TODO support dot see http://www.keil.com/pack/doc/CMSIS/SVD/html/svd__outline_pg.html
+                if peripheral.description is None:
+                    peripheral.description = parent.description
```

Comment:
The problem with short circuiting is that both `None` and `0x00000000` are interpreted as false, therefore anywhere where o is a valid value we must still distinguish between 0 and `None`.